src/pcre2_compile.c

```diff
@@ -11005,7 +11051,7 @@ used in this code because at least one compiler gives a warning about loss of
 "const" attribute if the cast (PCRE2_UCHAR *)codestart is used directly in the
 function call. */
 
-if (errorcode == 0 && (re->overall_options & PCRE2_NO_AUTO_POSSESS) == 0)
+if (errorcode == 0 && (optim_flags & PCRE2_OPTIM_AUTO_POSSESS))
```

Comment:
In many cases the `== 0` and `!= 0` are removed. Is this a new coding style? We should eventually update all cases them, except for pointers.

Reply:
I'm just trying to make the code shorter and didn't see any reason to include `!= 0`. However, I can add it back, that isn't a problem.

I am a new contributor to this project and don't know all the local conventions yet.

Reply:
This is one of my "fetishes" -- I only ever use if (x) when x is a Boolean; if it's not I use if (x != 0) to emphasize that it is a numerical value. Similar to my use of if (pointer != NULL) to emphasize that the variable is a pointer (though in principle, NULL need not be zero, so this is more correct C).